arangod/VocBase/Methods/Collections.cpp

```diff
@@ -605,32 +605,170 @@ Collections::create(         // create collection
     bool enforceReplicationFactor,            // replication factor flag
     bool isNewDatabase, bool allowEnterpriseCollectionsOnSingleServer,
     bool isRestore) {
-  std::vector<std::shared_ptr<LogicalCollection>> results;
-  results.reserve(collections.size());
-  // This block is to be replaced by proper implementation
-  {
-    // In this rudimentary forward we only support single collection, as this is
-    // the only way it is used right now.
-    TRI_ASSERT(collections.size() == 1);
-    // Just plainly forward
-    auto& planCollection = collections.at(0);
-    auto parameters = planCollection.toCollectionsCreate();
-    std::shared_ptr<LogicalCollection> coll;
-    auto res = methods::Collections::create(
-        vocbase, options,
-        planCollection.name,            // collection name
-        planCollection.getType(),       // collection type
-        parameters.slice(),             // collection properties
-        createWaitsForSyncReplication,  // replication wait flag
-        enforceReplicationFactor,       // replication factor flag
-        /*isNewDatabase*/ false,        // here always false
-        coll, planCollection.isSystem);
+  // Let's first check if we are allowed to create the collections
+  ExecContext const& exec = options.context();
+  if (!exec.canUseDatabase(vocbase.name(), auth::Level::RW)) {
+    for (auto const& col : collections) {
+      events::CreateCollection(vocbase.name(), col.name, TRI_ERROR_FORBIDDEN);
+    }
+    return arangodb::Result(                             // result
+        TRI_ERROR_FORBIDDEN,                             // code
+        "cannot create collection in " + vocbase.name()  // message
+    );
+  }
+
+  // TODO: Discuss should this be Prod Assert?
```

Comment:
I think it would be safe to just make this an error. If the vocbase is dangling, from the user's view, it should just not exist anymore. If we by any way mange to get a request through here and the vocbase is dangling, responding with "database does not exist" is a safe, non-catastrophic response.

using `ADB_PROD_ASSERT` would on the other hand just crash the process, so we should only make this assertions if we made sure we cannot get a collection creation request through the api layers above and *want* to assert that our protections work.